### PR TITLE
config for specific assertionConsumerServiceUrl

### PR DESCRIPTION
**Pull request by domivds**

### CODE REVIEW

File: security/src/main/java/zone/cogni/lib/security/saml2/Saml2HttpConfigurer.java [hunk 1]
```diff
@@ -41,21 +47,53 @@ public class Saml2HttpConfigurer extends SecurityHttpConfigurer<Saml2HttpConfigu
...
+        log.info("Will patch assertionConsumerServicePath in Saml2WebSsoAuthenticationFilter with {}", assertionConsumerServicePath);
+        filter.setFilterProcessesUrl(assertionConsumerServicePath);
+        return filter;
+      }
+
+      private String getAssertionConsumerServicePath() {
+        try {
+          String path = new URL(assertionConsumerServiceUrl).getPath();
+          if (!path.startsWith(contextPath)) throw new RuntimeException("Path of AssertionConsumerServiceUrl does not start with " + contextPath);
+          return "/" + StringUtils.removeStart(path, contextPath);
```
Comment (alexey-lukashov) [inline security/src/main/java/zone/cogni/lib/security/saml2/Saml2HttpConfigurer.java]: if I understand correctly AssertionConsumerServiceUrl should be on the same contextPath as the application? both things are static and could be validated on service startup, no?

**Review by alexey-lukashov — APPROVED** [review]
I only have a small remark about checking the context path of the assertion service when starting the server instead of the get method

File: security/src/main/java/zone/cogni/lib/security/saml2/Saml2HttpConfigurer.java (hunk 1 above)
Comment (domivds) [reply security/src/main/java/zone/cogni/lib/security/saml2/Saml2HttpConfigurer.java]: indeed (if both are set ofc)
but they kindof are validated at startup, the ObjectPostProcessor is used during HttpSecurity init/build

**Review by christinaspachiou — APPROVED** [review]

File: security/src/main/java/zone/cogni/lib/security/saml2/Saml2HttpConfigurer.java [hunk 2]
```diff
@@ -41,21 +47,53 @@ public class Saml2HttpConfigurer extends SecurityHttpConfigurer<Saml2HttpConfigu
...
             .addFilterBefore(metadataFilter, Saml2WebSsoAuthenticationFilter.class)
             .addFilterAfter(this::patchAuthenticationObjectFilter, Saml2WebSsoAuthenticationFilter.class);
   }
 
+  private void checkAssertionConsumerServiceUrl(Saml2LoginConfigurer<HttpSecurity> httpSecuritySaml2LoginConfigurer) {
+    String assertionConsumerServiceUrl = saml2Properties.getAssertionConsumerServiceUrl();
+    if (StringUtils.isBlank(assertionConsumerServiceUrl) || assertionConsumerServiceUrl.endsWith("/{registrationId}")) return;
+
+    RelyingPartyRegistration registrationId = relyingPartyRegistrationRepository.findByRegistrationId(saml2Properties.getRegistrationId());
+    ObjectPostProcessor<Saml2WebSsoAuthenticationFilter> patchSaml2WebSsoAuthenticationFilterProcessor = new ObjectPostProcessor<>() {
```
Comment (natan-cox-cognizone) [inline security/src/main/java/zone/cogni/lib/security/saml2/Saml2HttpConfigurer.java]: I would probably put this in a separate method?

File: security/src/main/java/zone/cogni/lib/security/saml2/Saml2HttpConfigurer.java [hunk 3]
```diff
@@ -41,21 +47,53 @@ public class Saml2HttpConfigurer extends SecurityHttpConfigurer<Saml2HttpConfigu
...
+      }
+
+      private String getAssertionConsumerServicePath() {
+        try {
+          String path = new URL(assertionConsumerServiceUrl).getPath();
+          if (!path.startsWith(contextPath)) throw new RuntimeException("Path of AssertionConsumerServiceUrl does not start with " + contextPath);
+          return "/" + StringUtils.removeStart(path, contextPath);
+        }
+        catch (MalformedURLException e) {
+          throw new RuntimeException(e);
```
Comment (natan-cox-cognizone) [inline security/src/main/java/zone/cogni/lib/security/saml2/Saml2HttpConfigurer.java]: log a message with failure? Like `"invalid assertionConsumerServiceUrl '" + ... "'",`

File: security/src/main/java/zone/cogni/lib/security/saml2/Saml2Configuration.java [hunk 4]
```diff
@@ -52,11 +60,13 @@ public RelyingPartyRegistrationRepository relyingPartyRegistrationRepository() {
...
                                                                            .registrationId(properties.getRegistrationId());
-
     String baseUrl = StringUtils.defaultIfBlank(properties.getBaseUrl(), "{baseUrl}");
     String entityIdTemplate = StringUtils.firstNonBlank(properties.getEntityId(), defaultEntityId);
+    String assertionConsumerServiceUrlTemplate = StringUtils.firstNonBlank(properties.getAssertionConsumerServiceUrl(), defaultAssertionConsumerServiceLocation);
+    log.info("Using entityIdTemplate: {}", entityIdTemplate);
+    log.info("Using assertionConsumerServiceUrlTemplate: {}", assertionConsumerServiceUrlTemplate);
     rprBuilder.entityId(entityIdTemplate.replace("{baseUrl}", baseUrl))
-              .assertionConsumerServiceLocation(defaultAssertionConsumerServiceLocation.replace("{baseUrl}", baseUrl));
+              .assertionConsumerServiceLocation(assertionConsumerServiceUrlTemplate.replace("{baseUrl}", baseUrl));
```
Comment (natan-cox-cognizone) [inline security/src/main/java/zone/cogni/lib/security/saml2/Saml2Configuration.java:69]: poor man spel :)

**Review by natan-cox-cognizone — APPROVED** [review]

**Pull request merged**